### PR TITLE
Respect tied values in function call

### PR DESCRIPTION
**Pull request by nocturnalastro**
I added a loop to evaluate tied parameters before outputs are calculated.

### CODE REVIEW

**Comment by pllim** [comment]
This is supposed to fix an existing bug? If so, please link to the PR. Also, looks like a test would be in order.

File: astropy/modeling/core.py [hunk 1]
```diff
@@ -704,6 +704,13 @@ def __call__(self, *inputs, **kwargs):
         inputs, format_info = self.prepare_inputs(*inputs, **kwargs)
         parameters = self._param_sets(raw=True)
 
+
+        for index, tied in enumerate([self.tied[pname] for pname in self.param_names]):
+            if tied is not False:
+                parameters[index] = tied(self)
+
+        print(parameters,self.tied.values())
```
Comment (mhvk) [inline astropy/modeling/core.py]: Don't forget to remove the print statement...

**Comment by nocturnalastro** [comment]
Its a bug I noticed and after a quick search I can't see an issue for it.

**Comment by mhvk** [comment]
@nocturnalastro - Great that you found the bug! It would seem like one that might cause hard-to-track problems. But, to ensure it doesn't recur, could you write a test case for which the code currently fails but which works with your change?

Also, and this is more for people who really understand the modelling code (@pllim, @astrofrog, @nden) -- is this the logical place to make the change? Or might it be better to do it in `param_sets` or so?

**Comment by pllim** [comment]
I don't use that part of modeling much, so I can't comment. @nden is on vacation for 2 weeks, so it might be a while before you hear back from her. FYI.

**Comment by nden** [comment]
Originally constraints were meant to be supported only in fitting. When fitters evaluate the parameters all constraints are respected and this is part of the fitter code. There have been requests to make them available in model evaluation as well (#2265 mentions something about this but as far as I know there is no specific issue). In this sense it's not a bug but an enhancement. And it should be done consistently for all constraints. 

Perhaps the right place to do this is when (re)setting a parameter or a constraint. For example, setting a `tied` constraint will reset the tied parameter and reseting a parameter should reset any parameters which are tied to it. It is not always clear what to do. If a parameter value is outside the bounds should it be reset and to what? It'd be good to hear opinions.

**Comment by nocturnalastro** [comment]
Hmm I see what you mean.
Because I found my self getting muddled whilst trying to explain this I'll say that parameter *b* = *func*(*a* ). 
Perhaps bounds on tied parameters such as *b* can be ignored and we could assume that  parameter it *a* has sufficient bounds also that if a parameter becomes tied it also is no longer fixed. 
But yes I agree it would be better to the evaluation of *func* when setting parameter *a*, is it possible to know what parameters are within *func* though as the model is just passed in currently isn't it? (I haven't delved into the *tied* code) As it would probably become a pain to re-evaluate every tied parameter on every parameter (re)setting.

**Pull request closed without merging**